src/SegmentedControl/SegmentedControl.jsx

```diff
@@ -22,19 +46,23 @@ export class SegmentedControl extends React.Component {
       return;
     }
 
-    this.setState({selected: value});
+    // Update internal state only if being used as an uncontrolled component.
+    if (!this.props.value) {
+      this.setState({selected: value});
+    }
```

Comment:
I found parsing this confusing at first read... at some point it might worth making two different components instead of one that can be in either controlled mode or uncontrolled mode